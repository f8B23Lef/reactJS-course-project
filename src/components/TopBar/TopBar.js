import React, { useState } from 'react';
import { createPortal } from 'react-dom';
import Logo from '../Logo/Logo';
import Modal from '../../modals/Modal/Modal';
import AddMovieModal from '../../modals/AddMovieModal/AddMovieModal';
import './TopBar.scss';

export default function TopBar() {
  const [showModal, setShowModal] = useState(false);

  return (
    <div className='top-bar'>
      <Logo />
      <button
        type="button"
        className='add-movie__button'
        onClick={() => setShowModal(true)}
      >
        + ADD MOVIE
      </button>
      {showModal && createPortal(
        <Modal onClose={() => setShowModal(false)}>
          <AddMovieModal onClose={() => setShowModal(false)} />
        </Modal>,
        document.body,
      )}
    </div>
  );
}
