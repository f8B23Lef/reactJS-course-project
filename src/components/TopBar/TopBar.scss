@import '../../styles/index.scss';

.top-bar {
  display: flex;
  justify-content: space-between;
  padding: 20px 35px;

  .add-movie__button {
    padding: 15px;
    border: none;
    border-radius: 4px;
    color: $red;
    background-color: $translucent-grey;
    cursor: pointer;

    &:hover {
      background-color: $light-grey;
    }
  }
}
