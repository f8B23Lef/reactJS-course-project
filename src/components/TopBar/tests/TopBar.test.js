import React, { useState as useStateMock } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import TopBar from '../TopBar';

jest.mock('../../Logo/Logo', () => () => 'Logo');

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}));

describe('TopBar', () => {
  const setStateMock = jest.fn();

  beforeEach(() => {
    useStateMock.mockImplementation((init) => [init, setStateMock]);
  });

  it('renders TopBar snapshot', () => {
    const { asFragment } = render(<TopBar />);

    expect(asFragment(<TopBar />)).toMatchSnapshot();
  });

  it('finds "Add movie" button by text', () => {
    render(<TopBar />);

    expect(screen.getByText('+ ADD MOVIE')).toBeInTheDocument();
  });

  it('calls setState() with specific param on "Add movie" button click', () => {
    render(
      <BrowserRouter>
        <TopBar />
      </BrowserRouter>,
    );

    const button = screen.getByText('+ ADD MOVIE');
    fireEvent.click(button);

    expect(setStateMock).toHaveBeenCalledWith(true);
  });
});
