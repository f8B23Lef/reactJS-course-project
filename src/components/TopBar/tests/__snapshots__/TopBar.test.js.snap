// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar renders TopBar snapshot 1`] = `
<DocumentFragment>
  <div
    class="top-bar"
  >
    Logo
    <button
      class="add-movie__button"
      type="button"
    >
      + ADD MOVIE
    </button>
  </div>
</DocumentFragment>
`;
